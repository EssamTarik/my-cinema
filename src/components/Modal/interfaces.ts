export interface IProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
  onRequestClose: () => void;
}