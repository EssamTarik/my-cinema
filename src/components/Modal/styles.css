.full-screen-modal {
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1000;
  top: 0;
  left: 0;

  margin: 0;
  padding: 0;

  width: 100%;
  height: 100%;

  background-color: rgba(0, 0, 0, 0.7);
}
