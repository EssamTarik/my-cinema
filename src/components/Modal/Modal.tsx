import React from "react";
import { IProps } from "./interfaces";
import "./styles.css";

const Modal = ({ onRequestClose, children }: IProps) => (
  <div onClick={onRequestClose} className="full-screen-modal">
    <div
      onClick={(event) => event.stopPropagation()}
      className="modal__content"
    >
      {children}
    </div>
  </div>
);

export default Modal;
