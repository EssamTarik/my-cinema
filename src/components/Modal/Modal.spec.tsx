import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";

import Modal from "./Modal";

describe('Modal', () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <Modal onRequestClose={jest.fn()}><div /></Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('requests closing on click outside content', () => {
    const mockClose = jest.fn();

    const { container } = render(
      <Modal onRequestClose={mockClose}><div /></Modal>
    );
    const modal = container.querySelector('.full-screen-modal');
    fireEvent.click(modal!);
    expect(mockClose).toHaveBeenCalled();
  });

})

