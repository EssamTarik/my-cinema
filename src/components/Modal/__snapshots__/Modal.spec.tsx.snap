// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<DocumentFragment>
  <div
    class="full-screen-modal"
  >
    <div
      class="modal__content"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
